src/components/sync-configurator/BaseSyncConfigurator.jsx

```diff
@@ -58,28 +68,61 @@ class BaseSyncConfigurator extends React.Component {
         },
     ];
 
+    componentDidMount = async () => {
+        const { groupFilterName, levelFilterName } = this.props;
+        const { filters } = this.state;
+
+        const groupFilterData = groupFilterName ? await this.getGroupFilterData() : null;
+        const levelFilterData = levelFilterName ? await this.getLevelFilterData() : null;
+
+        this.setState({ filters: { ...filters, groupFilterData, levelFilterData } });
+    };
+
     componentDidUpdate = prevProps => {
         const { extraFiltersState } = this.props;
         if (extraFiltersState !== prevProps.extraFiltersState) {
             this.setState({ filters: { ...this.state.filters, ...extraFiltersState } });
         }
     };
 
-    backHome = () => {
+    onBackHome = () => {
```

Comment:
Note (not a code change request): It's usually more declarative to use a verb for the method instead of replicating the event name/prefix. Then the code may read: `<Link onClick={this.goHome}`.

Reply:
Looks good suggestion for me, in this case we had without verb, handle and on. To make it all consistent I renamed all the methods to use the on but makes sense what you point.